src/prototype_room_market.js

```diff
@@ -169,15 +175,41 @@ Room.prototype.sendPowerOwnRooms = function() {
   }
 };
 
+Room.prototype.sendEnergyToMyRooms = function() {
+  if ((_.size(Memory.needEnergyRooms) > 0) && (_.size(Memory.canHelpRooms) > 0) &&
+    _.includes(Memory.canHelpRooms, this.name) && !_.includes(Memory.needEnergyRooms, this.name) &&
+    (this.terminal.store[RESOURCE_ENERGY] > config.terminal.minEnergyAmount)) {
+    const diffNeedEnergyRooms = _.difference(Memory.needEnergyRooms, Memory.canHelpRooms);
+    const needEnergyRooms = _.size(diffNeedEnergyRooms) > 0 ? diffNeedEnergyRooms : Memory.needEnergyRooms;
+    const myRoom = _.shuffle(needEnergyRooms)[0];
```

Comment:
Why not send to the closest one? It's cheaper

Reply:
i did that before, but than not all rooms get the energy they need, only the closest to the best farming, and i wannted to push as much in the controllers of all rooms as i could